bravado/requests_client.py

```diff
@@ -25,12 +29,15 @@ class Authenticator(object):
     """
 
     def __init__(self, host):
+        # type: (str) -> None
         self.host = host
 
     def __repr__(self):
-        return u"%s(%s)" % (self.__class__.__name__, self.host)
+        # type: () -> str
+        return "%s(%s)" % (self.__class__.__name__, self.host)
```

Comment:
I don't think this is ok, you're changing the return type on Python 2. I assume this was `unicode` on purpose.

Reply:
This change has been made because `__repr__` should be returning a `str`, while we're returning `unicode`.
Now we have two alternatives:
 1) fix the code (as in the PR), the type will be correct but we could have failures related to formatting (NOTE: this is technically possible if `self.host` is `unicode` and it shouldn't )
 2) ignore typing for this specific case

NOTE: In `Authenticator.matches` we ensure that `self.host` is the same of the hostname extracted from the url and `reveal_type(urlparse.urlsplit('').hostname)` tells us that `hostname` will be `str`

Given that I would say that the change should be safe

Reply:
Sounds good!